Remove the params option from jenkns_plugin and yum_repository

abadger opened the pull request:
It was decided that these options which override Ansible module options
from a generic, unchecked dict are an antipattern for Ansible Modules
and must be removed:

https://meetbot.fedoraproject.org/ansible-meeting/2017-09-28/ansible_dev_meeting.2017-09-28-15.00.log.html

Fixes #30874

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
* jenkins_plugin
* yum_repository

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
devel
```

samdoran commented on lib/ansible/modules/packaging/os/yum_repository.py:
```diff
@@ -391,6 +385,8 @@
   - The repo file will be automatically deleted if it contains no repository.
   - When removing a repository, beware that the metadata cache may still remain
     on disk until you run C(yum clean all). Use a notification handler for this.
+  - "The C(params) parameter was removed in Ansible-2.5 due to circumventing Ansible's parameter
```
Maybe just say 'Ansible 2.5' rather than having the dash.

abadger replied:
Fixed.

abadger committed:
Remove the params option from jenkns_plugin and yum_repository

It was decided that these options which override Ansible module options
from a generic, unchecked dict are an antipattern for Ansible Modules
and must be removed:

https://meetbot.fedoraproject.org/ansible-meeting/2017-09-28/ansible_dev_meeting.2017-09-28-15.00.log.html

Fixes #30874

abadger committed:
Add params removal to changelog
